builder/dockerfile/internals.go

```diff
@@ -380,9 +380,11 @@ func hostConfigFromOptions(options *types.ImageBuildOptions) *container.HostConf
 	// provided to the daemon as it doesn't support them.
 	//
 	// This is in line with what the ContainerCreate API endpoint does.
-	networkMode := options.NetworkMode
-	if networkMode == "" || networkMode == network.NetworkDefault {
-		networkMode = runconfig.DefaultDaemonNetworkMode().NetworkName()
+	var networkMode string
+	if options.NetworkMode == "" || options.NetworkMode == network.NetworkDefault {
+		networkMode = networkSettings.DefaultNetwork
+	} else {
```

Comment:
Why did you introduce this else instead of keeping the original code as it was?

Reply:
oh! I think this was from a previous iteration; I can change it back 👍